src/components/Cards/Cards.test.jsx

```diff
@@ -1,51 +1,80 @@
 import React from 'react';
 import { render, screen, cleanup } from '@testing-library/react';
-import Cards from '../Cards.component';
-import { testingMock } from '../mock.data';
+import { BrowserRouter } from 'react-router-dom';
+import Cards from './Cards.component';
+import { mockData } from './mock.data';
+import '@testing-library/jest-dom';
 
 afterEach(() => {
   cleanup();
 });
 
 test('renders Cards component without crashing', () => {
-  render(<Cards />);
+  render(
+    <BrowserRouter>
+      <Cards />
+    </BrowserRouter>
+  );
   const cardsElement = screen.getByTestId('cards');
   expect(cardsElement).toBeInTheDocument();
 });
 
 test('renders Welcome title', () => {
-  render(<Cards />);
+  render(
+    <BrowserRouter>
+      <Cards />
+    </BrowserRouter>
+  );
   const cardsElement = screen.getByTestId('welcome-header');
   expect(cardsElement).toHaveTextContent('Welcome to the Challenge!');
 });
 
 test('renders mock data when no props are passed', () => {
-  render(<Cards />);
+  render(
+    <BrowserRouter>
+      <Cards />
+    </BrowserRouter>
+  );
   const cardsElements = screen.getAllByTestId('card');
   expect(cardsElements.length).toBe(25);
 });
 
 test('renders mock data when props are undefined', () => {
-  render(<Cards items={undefined} />);
+  render(
+    <BrowserRouter>
+      <Cards items={undefined} />
+    </BrowserRouter>
+  );
   const cardsElements = screen.getAllByTestId('card');
   expect(cardsElements.length).toBe(25);
 });
 
 test('renders mock data when props are null', () => {
-  render(<Cards items={null} />);
+  render(
+    <BrowserRouter>
+      <Cards items={null} />
+    </BrowserRouter>
+  );
   const cardsElements = screen.getAllByTestId('card');
   expect(cardsElements.length).toBe(25);
 });
 
-test('renders no cards when props array is empty', () => {
-  render(<Cards items={[]} />);
-  const cardsElements = screen.queryByTestId('card');
-  expect(cardsElements).not.toBeInTheDocument();
+test('renders all the mocked amount of item cards when props array is empty', () => {
+  render(
+    <BrowserRouter>
+      <Cards items={[]} />
+    </BrowserRouter>
+  );
+  const cardsElements = screen.queryAllByTestId('card');
+  expect(cardsElements.length).toBe(mockData.items.length);
 });
 
 test('renders card with passed props', () => {
```

Comment:
This name is still confusing I dont think its reflecting what its doing 

I suggest

Cards render the number of cards stored in the mock data object